src/amber.cc

```diff
@@ -51,16 +51,16 @@ amber::Result Amber::Parse(const std::string& input, amber::Recipe* recipe) {
   return {};
 }
 
-amber::Result Amber::Execute(const amber::Recipe* recipe, Options* opts) {
-  ShaderMap map;
-  return ExecuteWithShaderData(recipe, opts, map);
-}
-
-amber::Result Amber::ExecuteWithShaderData(const amber::Recipe* recipe,
-                                           Options* opts,
-                                           const ShaderMap& shader_data) {
+// Create an engine initialize it, and check the recipe's requirements.
+// Returns a failing result if anything fails.  Otherwise pass the created
+// engine out through |engine_ptr| and the script via |script|.
```

Comment:
Add a:

The |script| pointer should not be freed.

or something?

Reply:
Better comment added.